src/main/java/hudson/remoting/forward/PortForwarder.java

```diff
@@ -134,9 +134,10 @@ public static ListeningPort create(VirtualChannel ch, final int acceptingPort, F
 
         return ch.call(new Callable<ListeningPort,IOException>() {
             public ListeningPort call() throws IOException {
+                final Channel channel = getOpenChannelOrFail(); // We initialize it early, so the forwarder won's start its daemon if the channel is shutting down
```

Comment:
There is no sense in starting daemon when we cannot return the port back, right?

Reply:
I agree.